psgo.go

```diff
@@ -744,3 +808,53 @@ func processHGROUP(p *process.Process, ctx *psContext) (string, error) {
 func processState(p *process.Process, ctx *psContext) (string, error) {
 	return p.Status.State, nil
 }
+
+func readMappings(path string) ([]IDMap, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot open %s", path)
+	}
+	defer file.Close()
+
+	mappings := []IDMap{}
+
+	buf := bufio.NewReader(file)
+	for {
+		line, _, err := buf.ReadLine()
+		if err != nil {
+			if err == io.EOF {
+				return mappings, nil
+			}
+			return nil, errors.Wrapf(err, "cannot read line from %s", path)
+		}
+		if line == nil {
+			return mappings, nil
+		}
+
+		containerID, hostID, size := 0, 0, 0
+		if _, err := fmt.Sscanf(string(line), "%d %d %d", &containerID, &hostID, &size); err != nil {
+			return nil, errors.Wrapf(err, "cannot parse %s", string(line))
+		}
+		mappings = append(mappings, IDMap{ContainerID: containerID, HostID: hostID, Size: size})
+	}
+}
+
+// SetMappings initialize the mappings in OPTS to the user namespace for the specified process.
+// If the pid 0 is passed, then the mappings for the current process are used.
+func SetMappings(opts *JoinNamespaceOpts, pid int) error {
```

Comment:
I think we can avoid exposing this function by replacing the opts parameter of `JoinNamespaceAndProcessInfoWithOptions()` with a boolean.

Reply:
I'd also prefer to make the opts a return value rather than a pointer.

Reply:
I think we should allow the caller to specify any mapping.  It can be helpful with nested namespaces, and also if we are joining the namespace of a different process.

If you prefer the return value, I can do that, I've used the pointer so that if we expand the opts structure in future, it can still be initialized in a different way

Reply:
> I think we should allow the caller to specify any mapping. It can be helpful with nested namespaces, and also if we are joining the namespace of a different process.

Sounds good, I trust your judgement.

> If you prefer the return value, I can do that, I've used the pointer so that if we expand the opts structure in future, it can still be initialized in a different way

Can we make it a method then?

```golang
func (opts *JoineNameSpaceOpts) SetMapping(pid int) error {}
```

Reply:
I've changed it to be a `bool` and it gets automatically filled with the current namespace, so we can have both features (fill an arbitrary mapping and still be able to fill it automatically for the most common case)